### PR TITLE
Monitor more info with grafana

### PR DESCRIPTION
**Pull request by 0xakk0r0kamui**

### CODE REVIEW

File: chain/monitor.go [hunk 1]
```diff
@@ -105,9 +114,27 @@ func (r *Reporter) clearRequestsPerPeer() {
 	defer r.requestsPerPeer.Unlock()
 	for key := range r.requestsPerPeer.m {
 		delete(r.requestsPerPeer.m, key)
+		delete(r.requestsPerPeer.lm, key)
 	}
 }
 
+func (r *Reporter) pushDataToGrafana() {
+	timestep := 5 * time.Second
+	for ; true; <-time.Tick(timestep) {
```
Comment (0xbunyip) [inline chain/monitor.go]: use time.NewTicker

File: chain/monitor.go [hunk 2]
```diff
@@ -105,9 +114,27 @@ func (r *Reporter) clearRequestsPerPeer() {
...
+		if r.gralog == nil {
+			continue
+		}
+		r.requestCounts.Lock()
+		for k, v := range r.requestCounts.m {
+			c := v - r.requestCounts.lm[k]
```
Comment (0xbunyip) [inline chain/monitor.go]: we should stop clearing lm for more stable values
we can also remove requestCounts.m

File: route/manager.go [hunk 3]
```diff
@@ -65,7 +68,8 @@ func NewManager(
 			p,
 			supportShards,
 		),
-		host: h,
+		host:   h,
+		gralog: gl,
```
Comment (0xbunyip) [inline route/manager.go]: no use?

File: process/pubsub.go [hunk 4]
```diff
@@ -139,33 +144,41 @@ func (pubsub *PubSubManager) SubKnownTopics(fromInside bool) error {
...
-// 			pubsub.pubsubInfo.Locker.Unlock()
-// 		}
-// 		if event.Type == p2pPubSub.PeerLeave {
-// 			pubsub.pubsubInfo.Locker.Lock()
-// 			pubsub.pubsubInfo.Info[subs.Topic()] = deletePeerIDinSlice(event.Peer, pubsub.pubsubInfo.Info[subs.Topic()])
-// 			pubsub.pubsubInfo.Locker.Unlock()
-// 		}
-// 	}
-// }
+func (pubsub *PubSubManager) WatchingSubs(subs *p2pPubSub.Subscription) {
```
Comment (0xbunyip) [inline process/pubsub.go]: unused?

File: chain/monitor.go (hunk 1 above)
Comment (0xakk0r0kamui) [reply chain/monitor.go]: Done

File: chain/monitor.go (hunk 2 above)
Comment (0xakk0r0kamui) [reply chain/monitor.go]: done

File: process/pubsub.go [hunk 5]
```diff
@@ -139,33 +144,41 @@ func (pubsub *PubSubManager) SubKnownTopics(fromInside bool) error {
...
+		if event.Type == p2pPubSub.PeerJoin {
+			pubsub.Info.Locker.Lock()
+			pubsub.Info.Info[tp] = append(pubsub.Info.Info[tp], event.Peer)
+			if pubsub.GraLog != nil {
+				pubsub.GraLog.Add(fmt.Sprintf("total_%s", msgType), len(pubsub.Info.Info[tp]))
+			}
+			pubsub.Info.Locker.Unlock()
+		}
+		if event.Type == p2pPubSub.PeerLeave {
+			pubsub.Info.Locker.Lock()
```
Comment (0xbunyip) [inline process/pubsub.go]: do we need Info.Locker when no other goroutine access it?

File: process/pubsub.go (hunk 4 above)
Comment (0xakk0r0kamui) [reply process/pubsub.go]: Needed test before use

File: route/manager.go (hunk 3 above)
Comment (0xakk0r0kamui) [reply route/manager.go]: Yes, just add for someone who wanna monitor data of this struct

**Pull request merged**